lib/security/WSSecurity.js

```diff
@@ -75,16 +70,28 @@ WSSecurity.prototype.toXML = function() {
       "<wsse:Nonce EncodingType=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-soap-message-security-1.0#Base64Binary\">" + nonce + "</wsse:Nonce>";
   }
 
-  return "<wsse:Security " + (this._actor ? "soap:actor=\"" + this._actor + "\" " : "") +
-    (this._mustUnderstand ? "soap:mustUnderstand=\"1\" " : "") +
-    "xmlns:wsse=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd\" xmlns:wsu=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd\">" +
-    timeStampXml +
-    "<wsse:UsernameToken xmlns:wsu=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd\" wsu:Id=\"SecurityToken-" + created + "\">" +
-    "<wsse:Username>" + this._username + "</wsse:Username>" +
-    password +
-    (this._hasTokenCreated ? "<wsu:Created>" + created + "</wsu:Created>" : "") +
-    "</wsse:UsernameToken>" +
-    "</wsse:Security>";
+	return "<wsse:Security " + (this._actor ? "soap:actor=\"" + this._actor + "\" " : "") +
+		(this._mustUnderstand ? "soap:mustUnderstand=\"1\" " : "") +
+		"xmlns:wsse=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd\" xmlns:wsu=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd\">" +
+		timeStampXml +
+		"<wsse:UsernameToken xmlns:"+this._wsu.name+"=\""+this._wsu.path+"\" "+this._wsu.name+":"+this._wsu.id+"=\""+this._wsu.value+"\">" +
+		"<wsse:Username>" + this._username + "</wsse:Username>" +
+		password +
+		(this._hasTokenCreated ? "<wsu:Created>" + created + "</wsu:Created>" : "") +
+		"</wsse:UsernameToken>" +
+		"</wsse:Security>";
 };
 
 module.exports = WSSecurity;
+
+function getDate(d) {
```

Comment:
would you mind to move this function to the top of the file alongside all the other declarations so we don't have things implemented/positioned _after_ the call to `module.exports`?